### DISCUSSION TITLE
HeatConduction

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25649] [Q&A General]
All,

My input includes;
[./heat_conduction]
    type = HeatConduction # The declaration of heat conduction model
    variable = temperature
    block = 1
[../]

Then, I got an error message;
A 'HeatConduction' is not a registered object.
If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

In Makefile, I set as 
HEAT_CONDUCTION     := yes 

Please help,
S. Thomas

**Comment by GiudGiud** [comment]
did you recompile after enabling the heat_conduction module?
you may have to clean the repo before recompiling too (make clobberall)

**Reply by styyokuda** [reply]
Thank you, Guillaume.
I did "clobberall" and "make -j 6." But it still does not work.
I got a warning of
```
*** Warning, This code is deprecated and will be removed in future versions:
"registerAll" in Modules is deprecated. Please update your *App.C file(s) to call the new templated "registerAllObjects" method (e.g. ModulesApp::registerAllObjects<MyApp>(...))
```
I am ignoring it now but does it matter?
Please help. S. Thomas

**Reply by GiudGiud** [reply]
this warning can be ignored for now. But you'll have to fix it one day.
The message is fairly explicit, you need to change that registerAll call to a registerAllObject<.  your app name >
